x-pack/legacy/plugins/lens/public/metric_visualization_plugin/auto_scale.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+interface Props {
+  children: React.ReactNode | React.ReactNode[];
+}
+
+interface State {
+  scale: number;
+}
+
+export class AutoScale extends React.Component<Props, State> {
+  private child: Element | null = null;
+  private parent: Element | null = null;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = { scale: 1 };
+  }
+
+  setParent(el: Element | null) {
+    if (this.parent !== el) {
+      this.parent = el;
+      setTimeout(() => this.scale());
+    }
+  }
+
+  setChild(el: Element | null) {
+    if (this.child !== el) {
+      this.child = el;
+      setTimeout(() => this.scale());
+    }
+  }
+
+  scale() {
```

Comment:
If we are doing scaling based on DOM measurements here, we probably should set up some re-calculate logic if the container changes size.

Reply:
There isn't really a good API for detecting resize of a specific element. Right now, this logic relies on React repaints, so it resizes on render, but that's about it. We could pretty easily have it resize on screen resize. There appears to be some kind of onResize event for embeddables or at least dashboard grid items, but I don't know how deeply we want to integrate there.

I think I'm going to fix the logic so that we have a working commit that we can refer to in the future if we want, and then I'm going to rip it out like we talked about.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver is the go-to API for this kind of thing. It is also used by elastic-charts for the same thing and the polyfill is already bundled with Kibana: https://github.com/elastic/kibana/blob/master/package.json#L228